internal/wasm/memory.go

```diff
@@ -263,7 +264,16 @@ func (m *MemoryInstance) Grow(delta uint32) (result uint32, ok bool) {
 		return currentPages, true
 	} else { // We already have the capacity we need.
 		sp := (*reflect.SliceHeader)(unsafe.Pointer(&m.Buffer))
-		sp.Len = int(MemoryPagesToBytesNum(newPages))
+		if m.Shared {
+			// Use atomic write to ensure new length is visible across threads.
+			if math.MaxInt == math.MaxInt32 {
```

Comment:
TBH I guess I've never ever tested this branch

Reply:
I think you can use `atomic.StoreUintptr` here.